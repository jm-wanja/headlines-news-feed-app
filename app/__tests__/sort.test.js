import React from 'react';
import { mount, shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
// import SortPanel from '../client/components/newsly/SortPanel.js';

//test the design of the sort component
describe('sort panel component', () => {
    it('snapshot of SortPanel', () => {
        const sort = shallow(<SortPanel />);
        const tree = shallowToJSON(sort);
        expect(tree).toMatchSnapshot();
    })

    //test there is a div containing the radiogroup
    it('asserts there is a sort radiogroup', () => {
        const sort = shallow(<SortPanel />);
        expect(sort.find('#sort')).toBe.defined;
    })
    //test there is a top radio
    it('displays a top radio', () => {
        const sort = shallow(<SortPanel />);
        expect(sort.find('.top').text()).toEqual('<Radio />');
    })

    //test there is a latest radio
    it('displays a latest radio', () => {
        const sort = shallow(<SortPanel />);
        expect(sort.find('.latest').text()).toEqual('<Radio />');
    })

    //test there is a popular radio
    it('displays a popular radio', () => {
        const sort = shallow(<SortPanel />);
        expect(sort.find('.popular').text()).toEqual('<Radio />');
    })
})