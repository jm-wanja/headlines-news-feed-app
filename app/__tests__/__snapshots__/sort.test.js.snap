// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sort panel component snapshot of SortPanel 1`] = `
<div
  className="radioGroup"
>
  <RadioGroup
    container="div"
    name="sort"
    value="opt1"
  >
    <Radio
      className="top"
      ripple={true}
      value="opt1"
    >
      Top
    </Radio>
    <Radio
      className="latest"
      ripple={true}
      value="opt2"
    >
      Latest
    </Radio>
    <Radio
      className="popular"
      ripple={true}
      value="opt3"
    >
      Popular
    </Radio>
  </RadioGroup>
</div>
`;
